addons/udes_stock/models/stock_picking.py

```diff
@@ -309,6 +310,40 @@ def action_done(self):
         all_picks = picks | picks.mapped("u_next_picking_ids")
         all_picks.with_context(lock_batch_state=False)._trigger_batch_state_recompute()
 
+        # If operation type does not allow multiple lots in a single transfer then check for number
+        # of different lots in move lines and raise an error if more than one
+        restrict_multi_lot_pickings = self.filtered(
+            lambda p: p.picking_type_id.u_restrict_multi_lot_pickings
+        )
+
+        for pick in restrict_multi_lot_pickings:
+            # Get the move lines tracked by lot or by none
+            non_serial_move_line_ids = pick.move_line_ids.filtered(
+                lambda ml: ml.product_id.tracking != "serial"
+            )
+
+            if non_serial_move_line_ids:
+                lot_count = 0
+
+                # Count the number of unique lots in the move lines
+                lots = non_serial_move_line_ids.mapped("lot_id")
+                lot_count += len(lots)
```

Comment:
Could we directly set lot_count = len(lots) to remove line 326?

Reply:
Yep good spot. Problem with tinkering with the code so much, will go amend this.

Reply:
Have now changed this.